### PR TITLE
update configure system document

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
Rewrite the configure system documentation.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/systems/system_config.py [hunk 1]
```diff
@@ -147,7 +147,15 @@ def validate_config_type(cls, values):
...
-                values["config"]["accelerators"] = [{"device": acc} for acc in system_alias_class.accelerators]
+                valid_accelerators = []
+                for device in system_alias_class.accelerators:
+                    valid_accelerators.extend(
+                        {"device": acc["device"], "execution_providers": acc.get("execution_providers")}
+                        for acc in values["config"]["accelerators"]
+                        if acc["device"].lower() == device.lower()
+                    )
+
+                values["config"]["accelerators"] = valid_accelerators or None
```
Comment (trajepl) [inline olive/systems/system_config.py:158]: Is it possible to remove this line?

File: olive/systems/system_config.py [hunk 2]
```diff
@@ -147,7 +147,15 @@ def validate_config_type(cls, values):
         if system_alias_class:
             values["type"] = system_alias_class.system_type
             if system_alias_class.accelerators:
-                values["config"]["accelerators"] = [{"device": acc} for acc in system_alias_class.accelerators]
+                valid_accelerators = []
+                for device in system_alias_class.accelerators:
+                    valid_accelerators.extend(
+                        {"device": acc["device"], "execution_providers": acc.get("execution_providers")}
+                        for acc in values["config"]["accelerators"]
```
Comment (trajepl) [inline olive/systems/system_config.py:154]: image
`values["config"]["accelerators"]` might be None, right? But None is not iterable

File: olive/systems/system_config.py [hunk 3]
```diff
@@ -147,7 +147,15 @@ def validate_config_type(cls, values):
         if system_alias_class:
             values["type"] = system_alias_class.system_type
             if system_alias_class.accelerators:
-                values["config"]["accelerators"] = [{"device": acc} for acc in system_alias_class.accelerators]
+                valid_accelerators = []
+                for device in system_alias_class.accelerators:
+                    valid_accelerators.extend(
+                        {"device": acc["device"], "execution_providers": acc.get("execution_providers")}
```
Comment (trajepl) [inline olive/systems/system_config.py:153]: If acc is not none, `execution_providers` must exist, right?

File: olive/systems/system_config.py (hunk 1 above)
Comment (guotuofeng) [reply olive/systems/system_config.py:158]: for AzureML alias, the accelerators including the ep must be provided.

File: olive/systems/system_config.py (hunk 2 above)
Comment (guotuofeng) [reply olive/systems/system_config.py:154]: for aml, the accelerators should not be None and must be provided.

File: olive/systems/system_config.py (hunk 3 above)
Comment (guotuofeng) [reply olive/systems/system_config.py:153]: acc cannot be None for aml, so do as ep

File: docs/source/tutorials/configure_systems.rst [hunk 4]
```diff
@@ -3,35 +3,59 @@
...
+
+* Native System: This is the normal system. All kinds of systems can be configured as native system. The **Isolated ORT system** can only be used as **target** system for model evaluation.
+* Managed System: Olive will manage the environment by installing the required packages from the :code:`requirements_file` or Dockerfile in the environment. Only **Python environment system**, **Docker system**, **AzureML system** support managed system.
+
+Most of time, the **host** and **target** are the same, but they can be different in some cases. For example, you can run a Pass on a local machine with a CPU and evaluate a Model on a remote machine with a GPU.
 
 Accelerator Configuration
 -------------------------
-For each **host** or **target**, it could specify multiple accelerators associated with it. Each accelerator could have the following attributes.
+For each **host** or **tartet**, it will represent list of accelerators that are supported by the system. Each accelerator is represented by a dictionary with the following attributes:
```
Comment (trajepl) [inline docs/source/tutorials/configure_systems.rst:21]: `tartet` -> `target`

**Review by devang-ml — APPROVED** [review]

**Pull request merged**